projects/go-lib/src/lib/components/go-button/go-button.component.ts

```diff
@@ -25,20 +35,44 @@ export class GoButtonComponent implements OnChanges {
 
   reset(): void { }
 
+  ngOnInit(): void {
+    this.setupButton();
+  }
+
   ngOnChanges(): void {
-    // 'alert' as a variant is depreciated and
-    // will be removed in a later version
-    const isNegative: boolean = [
-      'alert',
-      'negative'
-    ].includes(this.buttonVariant);
+    this.setupButton();
+    this.buttonLoader();
+  }
+
+  private setupButton(): void {
+    this.buttonVariant = this.buttonVariant === 'positive' ? 'primary' : this.buttonVariant;
 
     this.classObject = {
       'go-button--dark': this.useDarkTheme,
-      'go-button--loading': this.isProcessing,
-      'go-button--negative': isNegative,
-      'go-button--neutral': (this.buttonVariant === 'neutral'),
-      'go-button--positive': (this.buttonVariant === 'positive')
+      'go-button--loading': this.isProcessing
     };
+
+    this.classObject['go-button--' + this.buttonVariant] = true;
```

Comment:
Should we use string interpolation here?

```typescript
this.classObject[`go-button--${this.buttonVariant}`] = true;
```

Reply:
🤷‍♂ Sure.